transmart-rest-api/grails-app/services/org/transmartproject/rest/dataExport/ExportAsyncJobService.groovy

```diff
@@ -0,0 +1,105 @@
+package org.transmartproject.rest.dataExport
+
+import com.recomdata.transmart.domain.i2b2.AsyncJob
+import grails.transaction.Transactional
+import org.grails.web.json.JSONObject
+import org.quartz.JobKey
+import org.quartz.Scheduler
+import org.transmartproject.db.multidimquery.query.InvalidQueryException
+import org.transmartproject.db.user.User
+
+@Transactional
+class ExportAsyncJobService {
+
+    Scheduler quartzScheduler
+
+    static String jobType = 'DataExport'
+
+    /**
+     * Get the list of jobs to show in the jobs tab
+     */
+    List<AsyncJob> getJobList(User user) {
+        // TODO add time constraints for retrived jobs
+        if(user.admin) {
+            AsyncJob.findAllByJobType(jobType)
+        } else {
+            AsyncJob.findAllByUserIdAndJobType(user.username, jobType)
+        }
+    }
+
+    AsyncJob getJobByName(String jobName) {
+        def jobs = AsyncJob.findAllByJobName(jobName)
+        if(jobs.size() > 1) {
+            throw new InvalidQueryException("More than one job with a name '$jobName' found")
+        }
+        jobs[0]
+    }
+
+    String getDefaultJobName(String userName, Long jobId) {
+        userName + "-" + jobType + "-" + jobId
+    }
+
+    AsyncJob createNewJob(User user, String jobName = null) {
+        def jobStatus = JobStatus.STARTED
+
+        def newJob = new AsyncJob(lastRunOn: new Date())
+        newJob.jobType = jobType
+        newJob.jobStatus = jobStatus.value
+        newJob.jobName = jobName
+        newJob.userId = user.username
+        newJob.save()
+
+        if (!jobName?.trim()) {
+            newJob.jobName = getDefaultJobName(user.username, newJob.id)
+            newJob.save()
+        }
+
+        newJob
+    }
+    
+    /**
+     * Cancel a running job
+     */
+    def cancelJob(jobName, group = null) {
```

Comment:
This is not used for now, but we may want to have function like this at some point.